backend/app/notify/email.go

```diff
@@ -0,0 +1,242 @@
+	}
+	if err = testSMTPClient.Quit(); err != nil {
+		log.Printf("[WARN] failed to send quit command to %s:%d, %v", res.Host, res.Port, err)
+	}
+	if err := testSMTPClient.Close(); err != nil {
+		log.Printf("[WARN] can't close smtp connection, %v", err)
+	}
+	return &res, err
+}
+
+// Send email from request to address in settings
+func (e *Email) Send(ctx context.Context, req request) error {
+	log.Printf("[DEBUG] send notification via %s, comment id %s", e, req.comment.ID)
+	// TODO: decide where to get "to" email from
+	to := "test@localhost"
+	msg := e.buildMessageFromRequest(req, to)
+	err := repeater.NewDefault(5, time.Millisecond*250).Do(ctx, func() error {
+		return e.sendEmail(msg, to)
+	})
+	return err
+}
+
+// sendEmail sends message, prepared by Email.buildMessage
+func (e *Email) sendEmail(message, to string) error {
+	e.count++
+	e.sendMutex.Lock()
+	if e.SMTPClient == nil { // if client not set make new net/smtp
+		c, err := e.client()
+		if err != nil {
+			e.count--
+			e.sendMutex.Unlock()
+			return errors.Wrap(err, "failed to make smtp client")
+		}
+		e.SMTPClient = c
+	}
+
+	defer func() {
+		// count == 1 means that no one else waiting to send the message at this moment,
+		// e.SMTPClient is nil if Quit() call passed because it's closing connection as well
+		if e.count == 1 && e.SMTPClient != nil {
+			if err := e.SMTPClient.Close(); err != nil {
+				log.Printf("[WARN] can't close smtp connection, %v", err)
+			}
+			e.SMTPClient = nil
+		}
+		e.count--
+		e.sendMutex.Unlock()
+	}()
+
+	if err := e.SMTPClient.Mail(e.From); err != nil {
+		return errors.Wrapf(err, "bad from address %q", e.From)
+	}
+	if err := e.SMTPClient.Rcpt(to); err != nil {
+		return errors.Wrapf(err, "bad to address %q", to)
+	}
+
+	writer, err := e.SMTPClient.Data()
+	if err != nil {
+		return errors.Wrap(err, "can't make email writer")
+	}
+
+	buf := bytes.NewBufferString(message)
+	if _, err = buf.WriteTo(writer); err != nil {
+		return errors.Wrapf(err, "failed to send email body to %q", to)
+	}
+	if err = writer.Close(); err != nil {
+		log.Printf("[WARN] can't close smtp body writer, %v", err)
+	}
+
+	// count == 1 means that no one else waiting to send the message at this moment
+	if e.count == 1 {
+		if err = e.SMTPClient.Quit(); err != nil {
+			log.Printf("[WARN] failed to send quit command to %s:%d, %v", e.Host, e.Port, err)
+		} else {
+			e.SMTPClient = nil
+		}
+	}
+	return nil
+}
+
+func (e *Email) String() string {
+	return fmt.Sprintf("email: %s using '%s'@'%s':%d", e.From, e.Username, e.Host, e.Port)
+}
+
+//buildMessageFromRequest generates email message based on request
+func (e *Email) buildMessageFromRequest(req request, to string) (message string) {
+	subject := "New comment"
+	if req.comment.PostTitle != "" {
+		subject += fmt.Sprintf(" for \"%s\"", req.comment.PostTitle)
+	}
+	msg := bytes.Buffer{}
+	// we don't expect valid template to fail
+	_ = e.template.Execute(&msg, tmplData{
```

Comment:
I would check for an error anyway. Not sure in what cases template execution may fail but generally if called function returns an error the consumer should handle it